SmartDeviceLink/SDLManagerDelegate.h

```diff
@@ -47,6 +48,14 @@ NS_ASSUME_NONNULL_BEGIN
  */
 - (void)systemContext:(nullable SDLSystemContext)oldContext didChangeToContext:(SDLSystemContext)newContext;
 
+/**
+ * Called when the lifecycle manager detected a language mismatch. In case of a language mismatch the manager will try to change the apps registration by updating the lifecycle configuration to the specified language. If the app can support the specified language it should return an Object of SDLLifecycleConfigurationUpdate, otherwise it should return nil to indicate that the language is not supported.
+ *
+ * @param language The language of the connected head unit the manager is trying to update the configuration.
+ * @return An object of SDLLifecycleConfigurationUpdate if the head unit language is supported, otherwise nil to indicate that the language is not supported.
+ */
+- (nullable SDLLifecycleConfigurationUpdate *)managerWillUpdateLifecycleToLanguage:(SDLLanguage)language;
```

Comment:
I don't think I like the name of this method because this implies that it *will* like it or not, when it's actually asking "do you want to?"

`managerDidUpdateToLanguage:` I think is my preferred, because it did update to that language and now the dev can return something to update their config.

Reply:
I must disagree. It's following Apple's best practice of maning like they do in various delegates in UIKit.

The wording is: "The manager will update the lifecycle (configuration) to the specified language (because the language doesn't match the actual language of the head unit) unless nothing (nil) is returned to that the lifecycle can be updated to".

Reply:
I was seeking for an example of "will". Check this out [tableView:willSelectRowAtIndexPath:](https://developer.apple.com/documentation/uikit/uitableviewdelegate/1614943-tableview?language=objc) . However it's not the best example an old API...

Anyhow I think `managerShould...` makes much more sense.